cmake/external/gsl.cmake

```diff
@@ -3,11 +3,21 @@
 
 include(FetchContent)
 
-FetchContent_Declare(
+if(onnxruntime_USE_CUDA)
+  FetchContent_Declare(
     GSL
     URL https://github.com/microsoft/GSL/archive/refs/tags/v4.0.0.zip
     URL_HASH SHA1=cf368104cd22a87b4dd0c80228919bb2df3e2a14
-    )
+    PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/gsl/1064.patch
+  )
+else()
+  FetchContent_Declare(
+    GSL
+    URL https://github.com/microsoft/GSL/archive/refs/tags/v4.0.0.zip
+    URL_HASH SHA1=cf368104cd22a87b4dd0c80228919bb2df3e2a14
+  )
+endif()
```

Comment:
Correct me if I'm wrong, but the patch is safe to apply even when building with CUDA support turned off? What do you think about applying it unconditionally instead, so that both cuda-ful and cuda-free builds are built from "identical" source code?